packages/scanner/src/cli/scan/watchScan.ts

```diff
@@ -44,13 +47,53 @@ function isAncestorPath(ancestor: string, descendant: string): boolean {
   return !path.relative(ancestor, descendant).startsWith('..');
 }
 
+export type BatchedTelemetry = {
+  scanResults: ScanResults;
+  msElapsed: number;
+  msInserted: number;
+};
+
 export class Watcher {
   config?: TimestampedConfiguration;
   appmapWatcher?: chokidar.FSWatcher;
   appmapPoller?: chokidar.FSWatcher;
   configWatcher?: chokidar.FSWatcher;
+  batchedTelemetry: BatchedTelemetry[] = [];
+  sendPeriodMilliseconds = 5000;
+  interval: NodeJS.Timer;
+
+  constructor(private options: WatchScanOptions) {
+    this.interval = setInterval(() => {
+      this.sendScanResultsTelemetryInABatch();
+    }, this.sendPeriodMilliseconds);
+    this.interval.unref(); // else jest doesn't exit
+  }
 
-  constructor(private options: WatchScanOptions) {}
+  public sendScanResultsTelemetryInABatch() {
+    const dateNow = Date.now();
+    if (
+      this.batchedTelemetry &&
+      this.batchedTelemetry.length > 0 &&
+      dateNow - this.batchedTelemetry[0].msInserted > this.sendPeriodMilliseconds
+    ) {
+      const telemetryName = 'watchscan:completed';
+      const firstElement: BatchedTelemetry = this.batchedTelemetry[0];
+      const aggregatedScanResults: ScanResults = new ScanResults(
+        firstElement.scanResults.configuration,
+        firstElement.scanResults.appMapMetadata,
+        firstElement.scanResults.findings,
+        firstElement.scanResults.checks
+      );
+      aggregatedScanResults.clear();
```

Comment:
It's a little weird to be pulling out the first element to construct this object to then throw out the data (`clear`). Maybe the constructor should change to accept these arguments optionally? Or maybe we don't need this type?

Reply:
It is.  Fixed.